x86/src/x86.rs

```diff
@@ -238,8 +238,8 @@ impl X86 {
         let mut soonest = None;
         for (i, cpu) in self.cpus.iter().enumerate() {
             match cpu.state {
-                CPUState::Running => {}
-                CPUState::DebugBreak | CPUState::Error(_) | CPUState::Exit(_) => {
+                CPUState::Running | CPUState::Exit(_) => {}
```

Comment:
This area is a little weird.

Right now CPUState::Exit is used to mark "the process overall wants to terminate" (see the implementation of `exit_process`), and so the various places that consider CPU state, like this one, treat any CPU going into the exit state as a "quit the process now" marker.

Meanwhile, `new_cpu` (the function used to start a new thread) doesn't consider any "free" CPUs (as would be freed up by ExitThread) for reuse, hmm.

I think with this modification it's possible for this loop to not find a CPU that is ready to run (if all the CPUs got into the exit state), which would mean the `soonest.unwrap` below would fail.

I think, having typed all that out, calling this state "Exit" is the problem because it muddles "exit process" with "exit thread".  So I think what you should do instead is make a new CPUState, maybe `::Free`, that a terminated thread goes into when it's no longer in use, and then use that in ExitThread, retrowin32_thread_main, and new_cpu.